nomad/structs/config/artifact.go

```diff
@@ -223,6 +233,10 @@ func (a *ArtifactConfig) Validate() error {
 		return fmt.Errorf("decompression_size_limit must be < %d but found %d", int64(math.MaxInt64), v)
 	}
 
+	if a.DisableArtifactInspection == nil {
+		return fmt.Errorf("disable_artifact_inspection must be set")
+	}
```

Comment:
this really smells to me like we do not want a `*bool` if the value is required.